src/storage/mvcc/reader/point_getter.rs

```diff
@@ -151,15 +155,27 @@ impl<S: Snapshot> PointGetter<S> {
             }
         }
 
+        let mut can_be_cached_by_lock = true;
         match self.isolation_level {
             IsolationLevel::Si => {
                 // Check for locks that signal concurrent writes in Si.
-                self.load_and_check_lock(user_key)?;
+                self.load_and_check_lock(user_key, Some(&mut can_be_cached_by_lock))?;
```

Comment:
If there is a lock, it will return early because of `?` operator. Then if the lock cannot be cached (after ts), `can_be_cached` loses update.

Reply:
Addressed